### ISSUE TITLE
hyperparameter tuning

### ISSUE THREAD

**Issue by jeff-regier** [prob-ml/bliss#66]
[Optuna](https://optuna.org/) integrates with pytorch-lightning to automatically find good hyperparameters. We could add a `--tuning` flag to our main executable script that initiates the search.

- [x] don't pass trial to the encoder
- [x] add a script to the `bin` directory that runs the hyper-parameter search
- [x] search for the optimal learning rate
- [x] early stopping based on validation loss, not counter loss
- [ ] use multiple GPUs in parallel (may require solving #80 )
